core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -156,11 +156,18 @@ public void onTimelineChanged(Timeline timeline, @Nullable Object manifest, int
             return;
         }
 
-        if (player != null) {
+        if (player != null && adPlaybackState != null) {
```

Comment:
without knowing much of how this used to work - it looks like now the check is stricter than before: how does this additional condition influence things like L171 (`notifyAdvertStart(advertBreaks.get(adGroupIndex));`)?

Reply:
Interesting point. So this shouldn't actually affect anything and in retrospect I believe we should have updated some of the other checks to also include this. My rationale is that this `AdPlaybackState` would only be `null` if the adverts failed to load or this class has already been released, in both cases we shouldn't be performing a notification for adverts anyway.